src/Makefile

```diff
@@ -231,6 +231,7 @@ SIMU_OPTION += -DRANDOM_NUMBER=RNG_GNU_EXT
 ##CXXFLAG     = -g -fast
 # CXXFLAG    += -w1                                       # warning flags
 # OPENMPFLAG  = -fopenmp                                  # openmp flag
+# CXXFLAG    += -DGIT_SHA1="`git rev-parse HEAD`"         # record SHA1 in git
```

Comment:
- Also record the git branch by `git rev-parse --abbrev-ref HEAD`.
- Put this line somewhere else since it should be independent of the adopted compiler (e.g., below [here](https://github.com/gamer-project/gamer/blob/master/src/Makefile#L780)).
- Print a warning message if `git` doesn't work (instead of aborting the compilation)

Reply:
@xuanweishan 
- Since this is not really a compiler flag, I would suggest storing it in a separate variable (e.g., `GIT_INFO`) and adding it [here](https://github.com/gamer-project/gamer/blob/master/src/Makefile#L931). Also, remove the extra whitespace below [here](https://github.com/gamer-project/gamer/blob/master/src/Makefile#L884).
- If `git` returns any error, set both `GIT_SHA1` and `GIT_BRANCH` to `"Unknown"`.